### PR TITLE
Sections 8.7 - 8.10: synchronization, ordering, locking

### PR DESCRIPTION
**Pull request by spotluri**
fixing mislabled PEs in quiet.pdf - addressing github issue #81

### CODE REVIEW

**Comment by min-si** [comment]
@spotluri : Thanks for creating the PR. I've created issue #127  for the fortran description issue.

**Comment by min-si** [comment]
@spotluri : Looks I do not have permission to add label or project card. Could you please mark PR [#128](https://github.com/openshmem-org/specification/pull/128) for me ? Thanks.

**Comment by min-si** [comment]
@spotluri : Here is another minor fix PR [#129](https://github.com/openshmem-org/specification/pull/129). Could you please label/card it as well ?

**Comment by spotluri** [comment]
@min-si: i have added the cards and also merged them into the section PR. please review.

**Comment by min-si** [comment]
Thanks !

**Comment by min-si** [comment]
When I was updating the nonblocking Put/Get in the memory ordering section, I was a little confused whether `shmem_fence` ensures ordering of nonblocking Get operations. It is also not explained in the RMA section.
Specifically, does fence ensure ordering in any of the follow cases ? 
1. Gets issued to the same remote pe:
```
shmem_get_nbi(loc_a, rem_a,...,pe0);
shmem_fence();
shmem_get_nbi(loc_b, rem_b,...,pe0);
```
2. Gets issued to different remote pe:
```
shmem_get_nbi(loc_a, rem_a,...,pe0);
shmem_fence();
shmem_get_nbi(loc_b, rem_b,...,pe1);
```
3. Get and Put issued to the same pe:
```
shmem_get_nbi(loc_a, rem_a,...,pe0);
shmem_fence();
shmem_put_nbi(rem_b, loc_b,...,pe0);
```

**Comment by spotluri** [comment]
@min-si sorry for pitching in late. 

My understanding of the current spec is that fence does not order get operations. So, it does not guarantee  ordering in any of the three cases you mention. 

@shamisp / @manjugv do you remember if this gets were intentionally left out of fence ordering. 

A change to this will be a semantic change and we should create a regular ticket if we would like to make this change.

**Comment by spotluri** [comment]
Hi All, 

Attached is a PDF of sections on synchronization, ordering, locking, now 8.8-8.11. I have merged all PRs and addressed issues raised until now. I have rebased it to the ToT of osh_spec_next. 

[section-sync-review-1.4.pdf](https://github.com/openshmem-org/specification/files/1283061/section-sync-review-1.4.pdf)

@min-si, @spophale, @shamisp, @manjugv, @abouteiller, can you please give your approval (thumbs up) on the draft. The deadline is Sept 7.

**Comment by manjugv** [comment]
> @min-si sorry for pitching in late.
> 
> My understanding of the current spec is that fence does not order non-blocking get operations. So, it does not guarantee ordering in any of the three cases you mention.
> 
> @shamisp / @manjugv do you remember if gets were intentionally left out of fence ordering.
> 
@min-si @spotluri That is correct. The fence does not order the nonblocking `Get` operations. This is to keep consistent with the blocking `Get` semantics.

**Comment by manjugv** [comment]
@spotluri You might have to remove Get from the commit, otherwise 👍

**Comment by jdinan** [comment]
How about adding a sentence to clarify that nonblocking gets are not ordered by fence?

**Comment by spotluri** [comment]
@manjugv, the draft does not add non-blocking gets to the list of operations that shmem_fence orders. non-blocking get have only been added to quiet. Am I missing an instance?

**Comment by spotluri** [comment]
objects to the same \ac{PE}. \FUNC{shmem\_fence} guarantees order of delivery,
-    not completion.
+    not completion. It does not guarantee order of delivery of non-blocking \GET{} routines.

**Comment by manjugv** [comment]
@spotluri I was referring to introduction paragraph main_spec.tex line 288/289. But, I guess that is OK.

**Comment by abouteiller** [comment]
Could we move table 5 and 6 above 8.9.1?

88:6: by a transfer from a remote PE -> by a PE
  rationale: could be a thread of the same PE

90:30: the following example uses shmem_fence in a C11 program.

Shouldn't the block of text in 8.10.2 page 91:35-40 also appear in shmem_fence ? (text about ordering of memory stores between multiple threads wrt shmem_quiet). 

92:30: "by the program user" -> in the user's program

**Comment by spotluri** [comment]
@abouteiller, @jdinan: thanks for the feedback. 

@abouteiller: I have merged your changes into the section PR. 

@jdinan: I have added a sentence clarifying that non-blocking gets are not ordered by fence.

@min-si, @spophale, @shamisp, @manjugv, @abouteiller, here is an updated draft for your review and approval

[section-sync-review-1.4-v2.pdf](https://github.com/openshmem-org/specification/files/1286209/section-sync-review-1.4-v2.pdf)

**Comment by shamisp** [comment]
👍

**Comment by manjugv** [comment]
👍

**Comment by min-si** [comment]
Sorry for my late response. I was on vacation in this week. The draft looks great to me.

File: content/shmem_fence.tex [hunk 1]
```diff
@@ -1,5 +1,5 @@
 \apisummary{
-    Assures ordering of delivery of \PUT{}, \acp{AMO}, and memory store routines
+    Assures ordering of delivery of \PUT{}, \ac{AMO}, memory store, and non-blocking \PUT{} routines
     to symmetric data objects.
```
Comment (spotluri) [inline content/shmem_fence.tex]: non-blocking -> nonblocking

File: content/shmem_fence.tex [hunk 2]
```diff
@@ -1,5 +1,5 @@
 \apisummary{
-    Assures ordering of delivery of \PUT{}, \acp{AMO}, and memory store routines
+    Assures ordering of delivery of \PUT{}, \ac{AMO}, memory store, and non-blocking \PUT{} routines
```
Comment (jdinan) [inline content/shmem_fence.tex]: Update "non-blocking" to "nonblocking"

File: content/shmem_fence.tex [hunk 3]
```diff
@@ -43,19 +43,29 @@
...
+    not the thread calling \FUNC{shmem\_fence}, the update must be made visible to the
+    calling thread according to the rules of the memory model associated with
+    the threading environment.
 }
 
 \begin{apiexamples}
 
 \apicexample
-    {The following \FUNC{shmem\_fence} example is for C11 programs: }
+    {The following example uses  \FUNC{shmem\_fence}  in a C11 program: }
```
Comment (nspark) [inline content/shmem_fence.tex]: `C11` --> `\Cstd[11]`

Comment (nspark) [reply content/shmem_fence.tex]: Similarly, this change should apply to `shmem_quiet.tex`.

Comment (BryantLam) [inline content/shmem_fence.tex]: \Cstd[11]

File: content/shmem_fence.tex [hunk 4]
```diff
@@ -43,19 +43,29 @@
...
+    guarantees completion of \PUT{}, \ac{AMO}, memory store, and non-blocking \PUT{} routines to
     symmetric data objects which makes the updates visible to all other
     \acp{PE}. 
     
     The \FUNC{shmem\_quiet} routine should be called if completion of \PUT{},
-    \acp{AMO}, and memory store routines to symmetric data objects is desired
+    \ac{AMO}, memory store, and non-blocking \PUT{} routines to symmetric data objects is desired
     when multiple remote \acp{PE} are involved.
+
+    In an \openshmem program with multithreaded \acp{PE}, it is the
```
Comment (abouteiller) [inline content/shmem_fence.tex]: @manjugv Did you have time to review this piece of text?

Comment (spotluri) [reply content/shmem_fence.tex]: i have closed the issue based on Manju's feedback

File: content/shmem_fence.tex (hunk 1 above)
Comment (spotluri) [reply content/shmem_fence.tex]: fixed

**Comment by nspark** [comment]
@spotluri I've opened https://github.com/spotluri/openshmem-specification/pull/11 to fix #175 and also noticed that you missed the editorial changes in https://github.com/spotluri/openshmem-specification/pull/9 that I opened after RCM1.

**Comment by spotluri** [comment]
@nspark: sorry, I did miss the post-RC1 changes PR. Merged it in now. 

 spotluri#11 looks good to me. Will wait on an approval from @jamesaross before merging it in.

**Pull request merged**